mllib/src/main/scala/org/apache/spark/ml/image/ImageSchema.scala

```diff
@@ -133,7 +133,13 @@ object ImageSchema {
    */
   private[spark] def decode(origin: String, bytes: Array[Byte]): Option[Row] = {
 
-    val img = ImageIO.read(new ByteArrayInputStream(bytes))
+    val img = try {
+      ImageIO.read(new ByteArrayInputStream(bytes))
+    } catch {
+      // Catch runtime exception because `ImageIO` may throw unexcepted `RuntimeException`.
+      // But do not catch the declared `IOException` (regarded as FileSystem failure)
+      case _: RuntimeException => null
```

Comment:
Shall we add a test?

Reply:
A test would be nice, though this is a pretty straightforward 'continuation' of the logic that's already below; ImageIO also returns null for bad images already. In practice I find lots of the libraries it uses internally to parse images will throw RuntimeExceptions on bad data though, so this LGTM

Reply:
Yeah, at best I would log 1 or N failures and then stop logging. While we don't have to do it here, indeed it might be puzzling to read 1000 images and always keep getting 998 or something with no warning?

Reply:
What about using an accumulator to keep track of the approximate number of rejected images? That way if peoples data goes "bad" they can at least see what's going?

Reply:
It might make sense to put the accumulator down in `readImages` when skipping invalid records is turned on.

Reply:
I suppose in other cases we have a new column for 'bad records' but not sure if that's overkill or what. I'm OK with just making this change here and doing more later as it already ignores some images that ImageIO doesn't read

Reply:
Oh, So there's a "dropInvalid" option. When  "dropInvalid" option turn off, the image datasource will still return invalid image row (but only the "origin" column have valid value). I think this feature is enough for user to debug something.